content/source/docs/plugin/framework/validation.html.md

```diff
@@ -0,0 +1,261 @@
+---
+layout: "extend"
+page_title: "Plugin Development - Framework: Validation"
+description: |-
+  How to validate configuration values using the provider development framework.
+---
+
+# Validation
+
+Practitioners implementing Terraform configurations desire feedback surrounding the syntax, types, and acceptable values. This feedback, typically referred to as validation, is preferably given as early as possible before a configuration is applied. Terraform supports validation for values in provider, resource, and data source configurations. The framework supports returning [diagnostics](./diagnostics.html) feedback on all of these, offering the ability to specify validations on an attribute, provider-defined type, and the entire provider, resource, or data source schema.
+
+~> **NOTE:** When implementing validation logic, configuration values may be [unknown](./types.html#unknown) based on the source of the value. Implementations must account for this case, typically by returning early without returning new diagnostics.
```

Comment:
Can we move this note down to where folks will be reading about implementation? I just want them to see it in context - I worry they'll forget about it if it stays up here :)

Reply:
Certainly makes sense! I'm worried about repeating this note up to 5 times (Creating Attribute Validators, Defining Type Validation, and the 3 schema validation sections). Do you have a suggestion for what do in this case?

Reply:
fair - I will leave this note here then. :)